### PR TITLE
Add rabbitmq applications and add couple of files for Kafka

### PR DESCRIPTION
**Pull request by ofertavivcx**

### CODE REVIEW

File: java-apps/java-rabbitmq-http-entry-point/src/main/java/rabbitmqhttpentrypoint/controller/RabbitMQController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import rabbitmqhttpentrypoint.consumer.Receiver;
+import rabbitmqhttpentrypoint.producer.Producer;
+import rabbitmqhttpentrypoint.properites.RelatedServicesProperties;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Home page hybrid.controller class.
+ */
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-http-entry-point/src/main/java/rabbitmqhttpentrypoint/controller/RabbitMQController.java:16]:
```suggestion
```

File: java-apps/java-rabbitmq-entry-point/src/main/java/rabbitmqentrypoint/producer/Producer.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Autowired
+    private RelatedServicesProperties relatedServicesProperties;
+
+    public Producer(RabbitTemplate rabbitTemplate) {
+        this.rabbitTemplate = rabbitTemplate;
+    }
+
+    public void send(byte[] message) throws Exception {
+        System.out.println("Sending message...");
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-entry-point/src/main/java/rabbitmqentrypoint/producer/Producer.java:21]: Maybe this should print what message is being sent?

```suggestion
        System.out.println("Sending message... <" + new String(message) + ">");
```

File: java-apps/java-rabbitmq-propagator/pom.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>service-discovery-demo-parent</artifactId>
+        <groupId>com.checkmarx.demo.service.discovery</groupId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>java-rabbitmq-propagator</artifactId>
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-propagator/pom.xml:12]:
```suggestion
    <modelVersion>4.0.0</modelVersion>

    <artifactId>java-rabbitmq-propagator</artifactId>
    <version>1.0.0-SNAPSHOT</version>
    
    <parent>
        <artifactId>service-discovery-demo-parent</artifactId>
        <groupId>com.checkmarx.demo.service.discovery</groupId>
        <version>1.0.0-SNAPSHOT</version>
    </parent>
```

File: java-apps/java-rabbitmq-propagator/run.bat [hunk 4]
```diff
@@ -0,0 +1 @@
+java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1011 -javaagent:"agent\cx-launcher.jar" -DcxAgentAutoUpgrade=no -Dcx.standalone=yes -Dcx.log.level=DEBUG -jar "target\java-rabbitmq-propagator.jar"
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-propagator/run.bat:1]:
```suggestion
java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8419 -javaagent:"agent\cx-launcher.jar" -DcxAgentAutoUpgrade=no -Dcx.standalone=yes -Dcx.log.level=DEBUG -jar "target\java-rabbitmq-propagator.jar"
```

File: java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/BaseConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+/*    @Bean
+    TopicExchange exchange() {
+        return new TopicExchange(relatedServicesProperties.getRabbitmqExchange());
+    }
+
+    @Bean
+    Binding binding(Queue queue, TopicExchange exchange) {
+        return BindingBuilder.bind(queue).to(exchange).with("foo.bar.#");
+    }*/
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/BaseConfiguration.java:32]: Leftovers?

File: java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/JavaRabbitMQPropagator.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class JavaRabbitMQPropagator {
+
+    public static void main(String[] args) {
+        SpringApplication.run(JavaRabbitMQPropagator.class, args);
+    }
+
+
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/JavaRabbitMQPropagator.java:13]:
```suggestion
```

File: java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/producer/Producer.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Autowired
+    private RelatedServicesProperties relatedServicesProperties;
+
+    public Producer(RabbitTemplate rabbitTemplate) {
+        this.rabbitTemplate = rabbitTemplate;
+    }
+
+    public void send(byte[] message) throws Exception {
+        System.out.println("Sending message...");
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/producer/Producer.java:21]:
```suggestion
        System.out.println("Sending message... <" + new String(message) + ">");
```

File: java-apps/java-rabbitmq-sink/pom.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>service-discovery-demo-parent</artifactId>
+        <groupId>com.checkmarx.demo.service.discovery</groupId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>java-rabbitmq-sink</artifactId>
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-sink/pom.xml:12]:
```suggestion
    <modelVersion>4.0.0</modelVersion>

    <artifactId>java-rabbitmq-sink</artifactId>
    <version>1.0.0-SNAPSHOT</version>
    
    <parent>
        <artifactId>service-discovery-demo-parent</artifactId>
        <groupId>com.checkmarx.demo.service.discovery</groupId>
        <version>1.0.0-SNAPSHOT</version>
    </parent>
```

File: java-apps/java-rabbitmq-sink/run.bat [hunk 9]
```diff
@@ -0,0 +1 @@
+java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1011 -javaagent:"agent\cx-launcher.jar" -DcxAgentAutoUpgrade=no -Dcx.standalone=yes -Dcx.log.level=DEBUG -jar "target\java-rabbitmq-sink.jar"
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-sink/run.bat:1]:
```suggestion
java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8420 -javaagent:"agent\cx-launcher.jar" -DcxAgentAutoUpgrade=no -Dcx.standalone=yes -Dcx.log.level=DEBUG -jar "target\java-rabbitmq-sink.jar"
```

File: java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/BaseConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+/*    @Bean
+    TopicExchange exchange() {
+        return new TopicExchange(relatedServicesProperties.getRabbitmqExchange());
+    }
+
+    @Bean
+    Binding binding(Queue queue, TopicExchange exchange) {
+        return BindingBuilder.bind(queue).to(exchange).with("foo.bar.#");
+    }*/
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/BaseConfiguration.java:32]: Leftovers?

File: java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/JavaRabbitMQSink.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class JavaRabbitMQSink {
+
+    public static void main(String[] args) {
+        SpringApplication.run(JavaRabbitMQSink.class, args);
+    }
+
+
```
Comment (yevgenykuz) [inline java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/JavaRabbitMQSink.java:13]:
```suggestion
```

File: java-apps/pom.xml [hunk 12]
```diff
@@ -25,6 +25,10 @@
         <module>java-kafka-http-entry-point</module>
         <module>java-kafka-propagator</module>
         <module>java-kafka-sink</module>
+        <module>java-rabbitmq-entry-point</module>
+        <module>java-rabbitmq-propagator</module>
+        <module>java-rabbitmq-sink</module>
+        <module>java-rabbitmq-http-entry-point</module>
```
Comment (yevgenykuz) [inline java-apps/pom.xml:31]:
```suggestion
        <module>java-rabbitmq-entry-point</module>
        <module>java-rabbitmq-http-entry-point</module>
        <module>java-rabbitmq-propagator</module>
        <module>java-rabbitmq-sink</module>
```

**Review by yevgenykuz — CHANGES_REQUESTED** [review]
Please consider my comments.

Other notes:
- No port for rabbitmq server?
- Can you please update Java README.md file too?

File: java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/BaseConfiguration.java (hunk 10 above)
Comment (ofertavivcx) [reply java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/BaseConfiguration.java:32]: yes,

Comment (ofertavivcx) [reply java-apps/java-rabbitmq-sink/src/main/java/rabbitmqsink/BaseConfiguration.java]: removed

File: java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/BaseConfiguration.java (hunk 5 above)
Comment (ofertavivcx) [reply java-apps/java-rabbitmq-propagator/src/main/java/rabbitmqpropagator/BaseConfiguration.java]: yes, removed

File: README.rst [hunk 13]
```diff
@@ -42,6 +42,16 @@ Port  App
 9003  Kafka (external listener)
 ====  =========================
 
+RabbitMQ
+~~~~~
```
Comment (yevgenykuz) [inline README.rst:46]:
```suggestion
~~~~~~~~
```

**Review by yevgenykuz — CHANGES_REQUESTED** [review]

**Review by yevgenykuz — APPROVED** [review]

**Pull request merged**